lib/practice_exercises.rb

```diff
@@ -1,13 +1,40 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  alph_str = string.downcase.gsub(/[^a-z\w]/,'') #substitute any non-letter or whitespace with nothing
```

Comment:
Note that this line here creates a new copy of the string... so you have O(n) space complexity.